cartridge/remote-control.lua

```diff
@@ -215,22 +239,36 @@ local function communicate(s)
             return true
         end
 
-        local ok, ret = pcall(rc_eval, code, args)
-        if ok then
-            reply_ok(s, sync, ret)
-            return true
-        elseif ffi.istype(error_t, ret) then
-            local code = ret.code
-            if code == nil and ret.errno ~= nil then
-                code = box.error.SYSTEM
+        local task = fiber.new(function(fiber_name, session)
+            acquire_socket(s)
+
+            fiber.name(fiber_name)
+            fiber.self().storage.session = session
+            local ok, ret = pcall(rc_eval, code, args)
+            if ok then
+                reply_ok(s, sync, ret)
+            elseif ffi.istype(error_t, ret) then
+                local code = ret.code
+                if code == nil and ret.errno ~= nil then
+                    code = box.error.SYSTEM
+                end
+                reply_err(s, sync, code, ret.message)
+            else
+                reply_err(s, sync, box.error.PROC_LUA, tostring(ret))
             end
-            reply_err(s, sync, code, ret.message)
-            return true
-        else
-            reply_err(s, sync, box.error.PROC_LUA, tostring(ret))
-            return true
-        end
 
+            if utils.table_count(vars.handlers) ~= 0 then
+                vars.handlers[s].storage.tasks[fiber.self():id()] = nil
+            end
```

Comment:
Why do you need this check?

First of all, if you really need to check that a table has at least one element you can call `if next(vars.handlers) ~= nil`.

But seems you need to check `vars.handlers[s] ~= nil`, so have I missed something? If yes, please add a comment.

Reply:
`drop_connection` applies `table.clear` on `vars.handlers`. So, I need `if next(vars.handlers) ~= nil` to check if `drop_connection` happened.